rspace/src/main/scala/coop/rchain/rspace/history/KeySegment.scala

```diff
@@ -0,0 +1,45 @@
+package coop.rchain.rspace.history
+
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+final case class KeySegment(value: ByteVector) extends AnyVal {
```

Comment:
Why is the value size check missing? Before that it was?

Reply:
I returned size check

Reply:
Why you reverted `extends AnyVal`? :thinking:

Reply:
I did it because I didn't find any benefit from the using value class instead of usual case class
![Screenshot from 2022-05-30 13-29-27](https://user-images.githubusercontent.com/90717554/171026948-bf8070e4-7b9e-481a-9240-d9ddbb01d61d.png)